plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -238,10 +220,8 @@ func (e *Elasticsearch) gatherNodeStats(url string, acc telegraf.Accumulator) er
 			"cluster_name": nodeStats.ClusterName,
 		}
 
-		if e.ClusterStats {
-			// check for master
-			e.isMaster = (id == e.catMasterResponseTokens[0])
-		}
+		// check for master
+		e.localNodeIsMaster = (id == e.masterNodeId)
```

Comment:
Perhaps, although presumably that shouldn't block this PR. In some regards though, as these master-only requests should only be coming from the master node, it's really not necessary to run with local=false. There's no need to have all nodes get data from all other nodes - there's no control over who's master, and why get all the redundant information. Or maybe I just don't understand the use case for local=false.